lib/services/route.dart

```diff
@@ -22,17 +20,18 @@ class HikingRoute {
     this.pointsOfInterest = pointsOfInterest;
     this.elevations = elevations;
     this.date = DateFormat('yyyy-MM-dd HH:mm').format(DateTime.now());
-    this.title = 'Sample title'; // TODO get region or custom info as title
+    setTitle('');
   }
 
   Future<String> buildTitle() async{
    List<Address> results = await Geocoder.local.findAddressesFromCoordinates(Coordinates(path[0].latitude, path[0].longitude));
-   return results[0].addressLine;
+   String street = (results[0].thoroughfare != null) ? results[0].thoroughfare : '';
+   String city = (results[0].locality.length + results[0].thoroughfare.length > 20) ? '\n' : ', ';
```

Comment:
exception when results[0].thoroghfare = null

Reply:
Cannot reproduce. Does it still occur with the current version?